File: frontend/src/types/comment.ts
import type { User } from './user'

export interface Comment {
  id: number
  postId: number
  parentId: number
  content: string
  likeCount: number
  status: number
  createdAt: string
}

export interface CommentWithUser extends Comment {
  author: User
  replyToUser?: User
  liked: boolean
  children?: CommentWithUser[]
}

export interface CommentCreateRequest {
  postId: number
  parentId?: number
  replyToUserId?: number
  content: string
}

File: backend/src/main/java/com/techcommunity/entity/Comment.java
package com.techcommunity.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("t_comment")
public class Comment {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long postId;

    private Long userId;

    private Long parentId;

    private Long replyToUserId;

    private String content;

    private Integer likeCount;

    private Integer status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
}
